binderhub/app.py

```diff
@@ -125,6 +142,17 @@ class BinderHub(Application):
         config=True
     )
 
+    docker_api_url = Unicode(
```

Comment:
docker_api_url looks like it would accept `tcp://1.2.3.4...`, is that true? I assume tcp sockets cannot be mounted in the way this is handled. If that's the case, `docker_socket` might be clearer, since it doesn't support any of the other forms. Plus, if it's docker_api_url, it should be `unix:///var/...`, and probably called `docker_host`, since that's what docker itself calls it (matching the DOCKER_HOST env).

Rather than docker_api_url, can we respect docker's own environment variables like `DOCKER_HOST` without our own logic? That should get all of docker's tcp/etc. support, right?

I'm not 100% sure how this would work here, since we want to talk to the same docker in both binderhub and the build, right? And docker environment variables can be paths that would presumably need to be mounted.

Reply:
Unfortunately we can't simply pass DOCKER_HOST around, since we have to do a hostPath volume mount if it's a unix socket. Since we currently only support doing this for unix sockets, I'm going to validate that it's a unix:/// url instead.